pandas/core/arrays/integer.py

```diff
@@ -442,17 +442,20 @@ def astype(self, dtype, copy: bool = True) -> ArrayLike:
             if incompatible type with an IntegerDtype, equivalent of same_kind
             casting
         """
-        from pandas.core.arrays.boolean import BooleanArray, BooleanDtype
+        from pandas.core.arrays.boolean import BooleanDtype
+        from pandas.core.arrays.string_ import StringDtype
 
         dtype = pandas_dtype(dtype)
 
         # if we are astyping to an existing IntegerDtype we can fastpath
         if isinstance(dtype, _IntegerDtype):
             result = self._data.astype(dtype.numpy_dtype, copy=False)
-            return type(self)(result, mask=self._mask, copy=False)
+            return dtype.construct_array_type()(result, mask=self._mask, copy=False)
         elif isinstance(dtype, BooleanDtype):
             result = self._data.astype("bool", copy=False)
-            return BooleanArray(result, mask=self._mask, copy=False)
+            return dtype.construct_array_type()(result, mask=self._mask, copy=False)
+        elif isinstance(dtype, StringDtype):
+            return dtype.construct_array_type()._from_sequence(self, copy=False)
```

Comment:
why is mask not passed in? and why is this using _from_sequence? e.g. this construction is *different* than from Integer/boolean

Reply:
we should either *always* use _from_sequence of never, mixing these is *very* confusing.

Reply:
That is because the BooleanArray constructor takes a mask, and the StringArray not. StringArray is not (yet) a masked array, so has a slightly different API.

(we need to revisit the full astype implementation anyway: https://github.com/pandas-dev/pandas/issues/22384)

Reply:
ok it would be great to do this next, as we are already creating a fair amount of tech debt here

Reply:
> to do this next

Making StringArray into a masked array, or the astype refactor? 
(the astype refactor actually first needs a concrete proposal, but I am planning to look into it after concat)

Reply:
> Making StringArray into a masked array

I thought this was *already* the case, I guess not.

Reply:
It used pd.NA as missing value, but it is not implemented as a masked array